spark/src/main/scala/ai/chronon/spark/JoinUtils.scala

```diff
@@ -319,8 +318,11 @@ object JoinUtils {
               case other     => other.toString // Keep other types (like Int) as they are
             }.toSet
 
-            // Form the final WHERE clause for injection
-            s"$groupByKeyExpression in (${valueSet.mkString(sep = ",")})"
+            if (valueSet.isEmpty) {
```

Comment:
`valueSet.nonEmpty`

Reply:
@hzding621 this should be isEmpty right? Because the value set will be empty if the notNullValues is empty and in this case we should add the filter as false?

Reply:
yeah this looks right now. I think I commented before seeing your commit https://github.com/airbnb/chronon/pull/853/commits/2f51a6208049b8889506d5332e242f637f544931